packages/site/src/hooks/MetamaskContext.tsx

```diff
@@ -79,57 +31,18 @@ export const MetaMaskProvider = ({ children }: { children: ReactNode }) => {
   }
 
   const [provider, setProvider] = useState<MetaMaskInpageProvider | null>(null);
-  const [state, dispatch] = useReducer(reducer, initialState);
+  const [error, setError] = useState<Error | undefined>();
 
   useEffect(() => {
     getSnapsProvider().then(setProvider).catch(console.error);
   }, []);
 
-  // Find MetaMask Provider and search for Snaps
-  // Also checks if MetaMask version is Flask
-  useEffect(() => {
-    /**
-     * Detect if the Snap is installed and set the result it in state.
-     */
-    async function detectSnapInstalled() {
-      dispatch({
-        type: MetamaskActions.SetInstalled,
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        payload: await getSnap(provider!),
-      });
-    }
-
-    /**
-     * Detect if the provider is Flask and set the result it in state.
-     */
-    async function checkIfFlask() {
-      dispatch({
-        type: MetamaskActions.SetIsFlask,
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        payload: await isFlask(provider!),
-      });
-    }
-
-    dispatch({
-      type: MetamaskActions.SetSnapsDetected,
-      payload: provider !== null,
-    });
-
-    if (provider) {
-      detectSnapInstalled().catch(console.error);
-      checkIfFlask().catch(console.error);
-    }
-  }, [provider]);
-
   useEffect(() => {
     let timeoutId: number;
 
-    if (state.error) {
+    if (error) {
       timeoutId = window.setTimeout(() => {
```

Comment:
```suggestion
      timeoutId = setTimeout(() => {
```

Reply:
Is it necessary to assign this to a mutable variable? Can't you use something like this instead?

```ts
useEffect(() => {
  if (condition) {
    const timeout = setTimeout(/* ... */);

    return () => {
      clearTimeout(timeout);
    };
  }
}, [/* ... */]);